### PR TITLE
fix: bugfix for popup persona

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MF-4723 MF-4726 MF-4900 MF-4906 MF-4911 MF-4918 MF-4919 MF-4921 MF-4926 MF-4702 MF-4532

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/plugins/Debugger/src/SiteAdaptor/components/ConnectionContent.tsx [hunk 1]
```diff
@@ -253,6 +253,8 @@ export function ConnectionContent(props: ConnectionContentProps) {
         }
     }, [])
 
+    const onPersonaSign = useCallback(() => {}, [])
```
Comment (guanbinrui) [inline packages/plugins/Debugger/src/SiteAdaptor/components/ConnectionContent.tsx]:
```suggestion
```

File: packages/web3-shared/evm/src/types/index.ts [hunk 2]
```diff
@@ -302,6 +305,8 @@ export interface MessageRequest {
 
 export type MessageResponse = JsonRpcResponse
 
+export type GasOptionLevel = GasOptionType | 'custom'
```
Comment (guanbinrui) [inline packages/web3-shared/evm/src/types/index.ts]: To add `GasOptionType.Custom`.

File: packages/mask/src/extension/popups/pages/Personas/components/PersonaHeader/UI.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        alignItems: 'center',
+        cursor: 'pointer',
+    },
+    avatar: {
+        marginRight: 4,
+        width: 30,
+        height: 30,
+    },
+    nickname: {
+        color: '#07101B',
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/pages/Personas/components/PersonaHeader/UI.tsx:38]: Use a themed color.

File: packages/mask/background/services/identity/persona/sign.ts [hunk 4]
```diff
@@ -19,6 +19,7 @@ export async function signWithPersona<T>(
     type: SignType,
     message: T,
     identifier?: ECKeyIdentifier,
+    source?: string,
```
Comment (guanbinrui) [inline packages/mask/background/services/identity/persona/sign.ts:22]:
```suggestion
```

File: packages/mask/background/services/identity/persona/sign.ts [hunk 5]
```diff
@@ -28,6 +29,7 @@ export async function signWithPersona<T>(
                 message: JSON.stringify(message),
                 requestID,
                 identifier: identifier?.toText(),
+                source,
```
Comment (guanbinrui) [inline packages/mask/background/services/identity/persona/sign.ts:32]:
```suggestion
                origin: location.origin,
```

File: packages/mask/background/services/identity/persona/sign.ts (hunk 4 above)
Comment (nuanyang233) [reply packages/mask/background/services/identity/persona/sign.ts:22]: No, This parameter is required for the signature page.

File: packages/mask/background/services/identity/persona/sign.ts (hunk 5 above)
Comment (guanbinrui) [reply packages/mask/background/services/identity/persona/sign.ts:32]: We can either detect the page when the method is being called.

Comment (nuanyang233) [reply packages/mask/background/services/identity/persona/sign.ts:32]: It doesn't work, because in the background.

Comment (guanbinrui) [reply packages/mask/background/services/identity/persona/sign.ts:32]: Then implement it in the front end.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**